crc/aarch64/crc_aarch64_dispatcher.c

```diff
@@ -34,6 +34,8 @@ DEFINE_INTERFACE_DISPATCHER(crc16_t10dif)
 	unsigned long auxval = getauxval(AT_HWCAP);
 	if (auxval & HWCAP_PMULL)
 		return PROVIDER_INFO(crc16_t10dif_pmull);
+#elif defined(__aarch64__)
```

Comment:
My suggestion is add it like 

`c
#ifndef __MACH__
unsigned long auxval=getauxval(AT_HWCAP);
if(auxval & HWCAP_PMULL)
    return PROVIDER_INFO(crc16_t10dif_pmull);
return PROVIDER_BASIC(crc16_t10dif);
#else
    return PROVIDER_INFO(crc16_t10dif_pmull);
#endif
`
And another thing I must confirm with you . If the transparent layer can be remove , I think this file should not be compiled in Apple Silicon Mac.

Reply:
I might not be answering correct question, but removing __MACH__ causes `getauxval undefined`.

Reply:
I have written dispatchers

Reply:
- aarch64's neon is spec
- my very first assumption was Apple would not ever sell aarch64 CPU without pmull. In this way I don't need dispatcher.

in above condition, removing 12575f52912ed1f62880b7fcf5fc5ed855020f2d solves "aarch64_multibinary.h issue" in workaround way.